charts/test-guide/templates/deployment.yaml

```diff
@@ -56,9 +56,18 @@ spec:
             timeoutSeconds: 5
             failureThreshold: 5
           readinessProbe:
-            httpGet:
-              path: /api/health/ready
-              port: {{ .Values.service.targetPort }}
+            exec:
+              command:
+                - /bin/sh
+                - -c
+                - |
+                  if curl -s http://localhost:{{ .Values.service.targetPort }}{{ .Values.service.readinessPath | default "/api/health/ready" }}; then
+                    exit 0
+                  elif curl -s http://localhost:{{ .Values.service.targetPort }}/api/health/live; then
+                    exit 0
+                  else
```

Comment:
this is a more complex solution than I expected. In this implementation it doesn't make much sense, as it would be enough to just check `.../live` for the same result.

If we want to do a more complex check if test.guide is in maintenance mode, we would need to check for http code 503 and a set `Retry-After` Header